openforcefield/topology/molecule.py

```diff
@@ -537,20 +1267,48 @@ def to_dict(self):
 
         """
         molecule_dict = OrderedDict()
-        molecule_dict['name'] = self.name
-        molecule_dict['atoms'] = [ atom.to_dict() for atom in self.atoms ]
-        molecule_dict['virtual_sites'] = [ vsite.to_dict() for vsite in self.virtual_sites ]
-        molecule_dict['bonds'] = [ bond.to_dict() for bond in self.bonds ] # TODO: How do we make sure bonds are serializable?
+        molecule_dict['name'] = self._name
+        ## From Jeff: If we go the properties-as-dict route, then _properties should, at
+        ## the top level, be a dict. Should we go through recursively and ensure all values are dicts too?
+        molecule_dict['atoms'] = [ atom.to_dict() for atom in self._atoms ]
+        molecule_dict['virtual_sites'] = [ vsite.to_dict() for vsite in self._virtual_sites ]
+        molecule_dict['bonds'] = [ bond.to_dict() for bond in self._bonds ]
         # TODO: Charges
         # TODO: Properties
+        # From Jeff: We could have the onerous requirement that all "properties" have to_dict() functions.
```

Comment:
Another possibility would be to have something like a `Conformation` or `Positions` object that knows how to serialize/deserialize itself with `to_dict()` and `from_dict()`. That way, we can reuse the same object for both `Molecule` and `Topology`.

Reply:
This is a really good idea -- Tracking as #138